core/src/apresolve.rs

```diff
@@ -82,30 +101,37 @@ impl ApResolver {
         let result = self.try_apresolve().await;
 
         self.lock(|inner| {
-            let data = match result {
-                Ok(data) => data,
-                Err(e) => {
-                    warn!("Failed to resolve access points, using fallbacks: {}", e);
-                    ApResolveData::default()
-                }
+            let (data, error) = match result {
+                Ok(data) => (data, None),
+                Err(e) => (ApResolveData::default(), Some(e)),
             };
 
-            inner.data.accesspoint = self.process_data(data.accesspoint);
-            inner.data.dealer = self.process_data(data.dealer);
-            inner.data.spclient = self.process_data(data.spclient);
+            inner.data.accesspoint = self.process_ap_strings(data.accesspoint);
+            inner.data.dealer = self.process_ap_strings(data.dealer);
+            inner.data.spclient = self.process_ap_strings(data.spclient);
+
+            if inner.data.is_any_empty() {
+                if let Some(error) = error {
+                    warn!(
+                        "Failed to resolve all access points, using fallbacks: {}",
+                        error
+                    );
+                } else {
+                    warn!("Failed to resolve all access points, using fallbacks");
```

Comment:
Nitpicking here, you could move this message up and print it always if `is_any_empty`, then print the error as an extra if there is one. Just a small DRY up.

Reply:
Good point, but in that case the it would generate two separate log events. Especially since this whole thing could run in parallel to other code it could in theory lead to a log like this:
```
[00:00:00] Failed to resolve all access points, using fallbacks
[00:00:01] Something unrelated also being logged
[00:00:01] Error message from the failed request, should come directly after the fallback message
```

I could do the general log if `is_any_empty` and then if there is an error do something like 
```rust
warn!("Access point resvole error: {}", e);
```

Let me know if you would prefer this

Reply:
I don't mind log messages in between. That's just a fact of life with threading.

Reply:
Should I still add at least the context like `warn!("Access point resolve error: {}", e);`, or just `warn!("{}", e);`?

Reply:
I like that extra bit of context.